contracts/sdjwt-verifier/src/types.rs

```diff
@@ -79,60 +86,97 @@ pub enum MathsOperator {
     EqualTo,
 }
 
+/// Validate the verified claims against the presentation request
 pub fn validate(
     presentation_request: PresentationReq,
     verified_claims: serde_json::Value,
-) -> Result<bool, SdjwtVerifierError> {
+    block_info: &BlockInfo,
+) -> Result<(), SdjwtVerifierResultError> {
     if let serde_json::Value::Object(claims) = verified_claims {
         for (key, criterion) in presentation_request {
-            if let Some(value) = claims.get(&key) {
+            // For every requested key, the value must be present in the verified claims
+            if key == CW_EXPIRATION && criterion == Criterion::Expires(true) {
+                if let Some(value) = claims.get(&key) {
+                    let exp = match value {
+                        serde_json::Value::String(exp) => {
+                            let expiration: Expiration =
+                                serde_json_wasm::from_str(exp).map_err(|_| {
+                                    SdjwtVerifierResultError::ExpirationStringInvalid(exp.clone())
+                                })?;
+                            expiration
+                        }
+                        _ => return Err(SdjwtVerifierResultError::CriterionValueTypeUnexpected),
+                    };
+                    if exp.is_expired(block_info) {
+                        return Err(SdjwtVerifierResultError::PresentationExpired(exp));
+                    }
+                } else {
+                    return Err(SdjwtVerifierResultError::DisclosedClaimNotFound(
+                        key.to_string(),
+                    ));
+                }
+            } else if let Some(value) = claims.get(&key) {
                 match (value, criterion) {
                     // matches the presentation value `p_val` with the criterion value `c_val`
                     (serde_json::Value::String(p_val), Criterion::String(c_val)) => {
                         if p_val != &c_val {
-                            return Ok(false);
+                            return Err(SdjwtVerifierResultError::CriterionValueFailed(key));
                         }
                     }
                     (serde_json::Value::Number(p_val), Criterion::Number(c_val, op)) => {
                         if p_val.is_u64() {
```

Comment:
No need to do if check and then unwrap(), i know that it won't panic, but it's better to write it like this
if let Some(p_val) = p_val.as_u64() {
}

Reply:
fixed in https://github.com/nymlab/AVIDA/tree/2d03896591111ec65986b15568dd90fb23bf0b3e